scripts/c_api_tests.sh

```diff
@@ -55,5 +59,8 @@ if [[ $(uname) == "Darwin" ]]; then
     nproc_bin="sysctl -n hw.logicalcpu"
 fi
 
-# Let's go parallel
-ARGS="-j$(${nproc_bin})" make test
+if [ "${WITH_FEATURE_GPU}" == "ON" ]; then
+    ctest --output-on-failure --test-dir "." --parallel "$(${nproc_bin})" --tests-regex ".*cuda.*"
+else
+    ctest --output-on-failure --test-dir "." --parallel "$(${nproc_bin})" --exclude-regex ".*cuda.*"
+fi
```

Comment:
The Makefile was used before.
We could also use it here
Is there a reason to do this this way now?

Reply:
This is equivalent than the make file, if you look at it, the makefile just calls ctest, here its 'cleaner' to pass arguments to cmake

```
test:
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --cyan "Running tests..."
	/home/linuxbrew/.linuxbrew/Cellar/cmake/3.24.3/bin/ctest --force-new-ctest-process $(ARGS)
```